src/cc65/segments.c

```diff
@@ -85,6 +96,86 @@ static Collection SegmentStack = STATIC_COLLECTION_INITIALIZER;
 
 
 
+void InitSegAddrSizes (void)
+/* Initialize the segment address sizes */
+{
+    InitCollection (&SegmentAddrSizes);
+}
+
+
+
+
+void DoneSegAddrSizes (void)
+/* Free the segment address sizes */
+{
+    SegAddrSize_t* A;
+    int I;
+    for (I = 0; I < (int)CollCount (&SegmentAddrSizes); ++I) {
+        A = CollAtUnchecked (&SegmentAddrSizes, I);
+        SB_Done (A->Name);
```

Comment:
`SB_Done()` frees the string _inside_ the StrBuf.  That line must free the entire StrBuf; use `FreeStrBuf()`.

Reply:
Hmmm, `A->Name` was not intended to be allocated separately from `A`.